sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -33,18 +34,30 @@ protected LogsIngestionClient()
         // If no concurrency count is provided for a parallel upload, default to 5 workers.
         private const int DefaultParallelWorkerCount = 5;
 
-        internal readonly struct BatchedLogs <T>
+        internal readonly struct BatchedLogs
         {
-            public BatchedLogs(int logsCount, BinaryData logsData)
+            public BatchedLogs(List<object> logs, BinaryData logsData)
             {
-                LogsCount = logsCount;
+                Logs = logs;
                 LogsData = logsData;
             }
 
-            public int LogsCount { get; }
+            public List<object> Logs { get; }
             public BinaryData LogsData { get; }
         }
 
+        internal readonly struct ProcessEventHandler
+        {
+            public ProcessEventHandler(bool shouldAbort, List<Exception> exceptions)
+            {
+                ShouldAbort = shouldAbort;
+                Exceptions = exceptions;
+            }
+
+            public bool ShouldAbort { get; }
+            public List<Exception> Exceptions { get; }
```

Comment:
Why is this a list - wouldn't there only be at most one exception?

Reply:
I think there can be an exception from the the Batch and then another one from the EventHandler? Maybe this is not a scenario though